app/pages/OrganizationEditPage.jsx

```diff
@@ -224,70 +245,94 @@ export class OrganizationEditPage extends React.Component {
         .then(data => {
           this.setState({
             resource: data.resource,
-            originalResource: data.resource,
-          });
-
-          const scheduleMap = {};
-          data.resource && data.resource.schedule && data.resource.schedule.schedule_days.forEach(day => {
-            scheduleMap[day.day] = day;
           });
-          this.setState({ scheduleMap });
         });
     }
   }
 
-  componentWillMount() {
-    this.props.router.setRouteLeaveHook(
-      this.props.route,
-      this.routerWillLeave,
-    );
-  }
-
   componentWillUnmount() {
     window.removeEventListener('beforeunload', this.keepOnPage);
   }
 
+  postServices = (servicesObj, promises) => {
+    if (!servicesObj) return;
+    const { resource } = this.state;
+    const newServices = [];
+    Object.entries(servicesObj).forEach(([key, value]) => {
+      const currentService = value;
+      if (key < 0) {
+        if (currentService.notesObj) {
+          const notes = Object.values(currentService.notesObj.notes);
+          delete currentService.notesObj;
+          currentService.notes = notes;
+        }
+
+        currentService.schedule = createFullSchedule(currentService.scheduleObj);
+        delete currentService.scheduleObj;
+
+        if (!_.isEmpty(currentService)) {
+          newServices.push(currentService);
+        }
+      } else {
+        const uri = `/api/services/${key}/change_requests`;
+        postNotes(currentService.notesObj, promises, { path: 'services', id: key });
+        delete currentService.notesObj;
+        postSchedule(currentService.scheduleObj, promises);
+        delete currentService.scheduleObj;
+        if (!_.isEmpty(currentService)) {
+          promises.push(dataService.post(uri, { change_request: currentService }));
+        }
+      }
+    });
+
+    console.log('newServices');
+    console.log(newServices);
```

Comment:
You still have some console.logs in here

Reply:
Nice catch! I just removed them, and I can merge as soon as Travis passes.